#Publish all joint states
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Joint trajectory controller
position_trajectory_controller:
  type: "position_controllers/JointTrajectoryController"
  joints:
    - joint_a1
    - joint_a2
    - joint_a3
    - joint_a4
    - joint_a5
    - joint_a6
  state_publish_rate: 50 # Defaults to 50
  action_monitor_rate: 20 # Defaults to 20

# eTaSl controller 1
etasl_controller:
  type: "etasl_ros_controllers/EtaslController"
  joints:
    - joint_a1
    - joint_a2
    - joint_a3
    - joint_a4
    - joint_a5
    - joint_a6
  input:
    names:
      - "tgt_x"
      - "tgt_y"
      - "tgt_z"
    types:
      - "Scalar"
      - "Scalar"
      - "Scalar"
  output:
    names:
      - "error_x"
      - "error_y"
      - "error_z"
      - "laser"
    types:
      - "Scalar"
      - "Scalar"
      - "Scalar"
      - "Vector"
  task_specification: "$(find etasl_ros_control_examples)/scripts/example_kuka.lua"

# eTaSl controller 2
etasl_controller_2:
  type: "etasl_ros_controllers/EtaslController"
  joints:
    - joint_a1
    - joint_a2
    - joint_a3
    - joint_a4
    - joint_a5
    - joint_a6
  task_specification: "$(find etasl_ros_control_examples)/scripts/example_kuka_2.lua"